### PR TITLE
feat: omnichain contract template

### PR DESCRIPTION
**Pull request by fadeev**
Adds a new task that generates the code for an omnichain smart contract.

The idea is that you would clone a template (the same for all use-cases), and then run a command to get you started with either omnichain smart contracts or CCM.

```
git clone https://github.com/zeta-chain/template swap
cd swap
yarn
npx hardhat omni swap destination:address recipient:bytes32 minOutput:uint256
```

This generates:

* A contract with the chosen name and the right fields in `message`
* A deploy task
* An interact task that also uses the right fields
* Next: tests

I think this could really streamline tutorials, because most code will be generated for you by the task. We generate only the code that is meant to be edited.

### CODE REVIEW

File: tasks/omni.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import { task, subtask } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import * as fs from "fs";
+import * as path from "path";
+import * as handlebars from "handlebars";
+import { upperFirst } from "lodash";
+
+async function processTemplates(
```
Comment (andresaiello) [inline tasks/omni.ts:8]: I just notice, in general, we are using arrow functions, any good reason to use this instead?

Comment (andresaiello) [reply tasks/omni.ts:8]: indeed, the main is arrow, I prefer to be consistent if both are ok

File: tasks/omni.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { task, subtask } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import * as fs from "fs";
+import * as path from "path";
+import * as handlebars from "handlebars";
+import { upperFirst } from "lodash";
```
Comment (andresaiello) [inline tasks/omni.ts:6]: does it worth including a lib just for this method?

File: tasks/omni.ts (hunk 1 above)
Comment (fadeev) [reply tasks/omni.ts:8]: Sure, I'll switch to arrows 👍

Comment (fadeev) [reply tasks/omni.ts]: https://github.com/zeta-chain/toolkit/pull/2/commits/36e999800fcdadb2e60c9d059f7d0276681d4c7d

File: tasks/omni.ts (hunk 2 above)
Comment (fadeev) [reply tasks/omni.ts]: https://github.com/zeta-chain/toolkit/pull/2/commits/36e999800fcdadb2e60c9d059f7d0276681d4c7d

File: package.json [hunk 3]
```diff
@@ -32,8 +32,10 @@
     "dotenv": "^16.0.3",
     "envfile": "^6.18.0",
     "ethers": "^5.4.7",
+    "handlebars": "^4.7.7",
     "hardhat": "^2.14.0",
     "hardhat-gas-reporter": "^1.0.8",
+    "lodash": "^4.17.21",
```
Comment (andresaiello) [inline package.json:38]: no longer needed

Comment (fadeev) [reply package.json]: https://github.com/zeta-chain/toolkit/pull/2/commits/92703e28bbde7c25269acc535f7a100611c318be

**Pull request merged**